build.sh

```diff
@@ -448,6 +448,10 @@ build_CoreLib()
         "$__ProjectRoot/dotnet.sh" publish --self-contained -r $__DistroRid -c $__BuildType -o "$__BinDir/crossgen2" "$__ProjectRoot/src/tools/crossgen2/crossgen2/crossgen2.csproj"
         rm $__BinDir/crossgen2/crossgen2.runtimeconfig.json
 
+        echo "Publishing tibcmgr for $__DistroRid"
```

Comment:
Why do we need a self-contained publish for this? Is the usage different from other managed tools like r2rdump?

The reason why we publish crossgen2 is because we need it to be runnable on the test machines that might not have a suitable `dotnet`.

Each self-contained publish produces 150 MB of stuff in /bin and CORE_ROOT that we then have to zip, send to all test machines, and unzip there - it easily adds 15 seconds to test runs. We have to be conservative about when we add such overhead.

Reply:
I published tibcmgr because we send it to the ASP.NET benchmarks server and don't control what dotnets they have. I can probably just publish this myself in the link-a-thon repo when we package up the payload so we don't do it every coreclr build.

Reply:
Single-exe branch doesn't get that much traffic, so it's fine if it helps a lot, but we'll probably have to think about this. Hopefully with the work around single exe we'll be able to cut self contained publish down by a fifth or so.

Reply:
I'll leave it for now because Ryan may need to re-build the benchmarks server payload and this will make life easier for him. I wanted to just jam crossgen2 and tibcmgr in the same folder but they're versioned differently - crossgen2 and its libraries are all versioned 1.0, whereas tibcmgr picks up the repo versioning and is 5.0, it also builds ILCompiler.ReadyToRun as 5.0, so crossgen and tibcmgr cannot share them. Once I understand how to make them line up we can stop publishing tibcmgr, and just put its binaries in the crossgen2 publish folder for the benchmark runs.